### PR TITLE
FST-83

### PR DESCRIPTION
**Pull request by ingrie**
**included in task** 

- Setup forms for each configuration - case, record, document and applicant
- create a model for keeping form data and a dto model for submitting data
- add the form fields corresponding with the fields in the Configuration model  
- dropdown fields for ones with FIXED_ARCHIVE_VALUE and text fields for the others
- add service for submitting forms 

**- not included**
- validation of fields
- support for properties in fields
- control section of the form to be used for control before submitting

### CODE REVIEW

File: src/features/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
+        <Box display="flex" position="relative" width={1} height={1}>
+            <Card className={classes.card} sx={{ maxWidth: 345 }}>
+                <CardContent>
+                    <Typography gutterBottom variant="h5" component="div">
+                        5 Skjema
+                    </Typography>
+                </CardContent>
+                <CardActions>
+                    <Button size="small" variant="outlined" component={RouterLink} to="/overview">Se integrasjoner</Button>
+                    <Button size="small" variant="outlined" component={RouterLink} to="/integration_configuration">Ny integrasjon</Button>
```
Comment (fsjovatsen) [inline src/features/dashboard/Dashboard.tsx:28]: Separate concepts with `/`.  This should probably be `/integration/configuration` or event (better?) `/integration/configuration/new`

File: src/features/integration/service/IntegrationService.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const getLatest = () => {
+    return http.get<Array<IFormData>>("/integration/configuration");
+};
+
+const getLatestByID = (id: any) => {
+    return http.get<Array<IFormData>>(`/integration/configuration/${id}/latest`);
+}
+
+const get = () => {
```
Comment (ingrie) [inline src/features/integration/service/IntegrationService.tsx:12]: will remove this duplicate

File: src/features/integration/util/MapToDto.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
+import IFormData from "../types/Form/FormData";
+import {IConfigurationDto} from "../types/ConfigurationDto";
+import {VALUE_BUILDER_STRATEGY} from "../types/ValueBuilderStrategy.enum";
+
+export function mapToDto(data: IFormData): IConfigurationDto {
```
Comment (fsjovatsen) [inline src/features/integration/util/MapToDto.ts:5]: What are you mapping to? Name the method according to what it is. `mapToDto` is to generic.

File: src/features/integration/IntegrationConfigurationForm.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            createNewConfiguration(dto);
+        }
+    });
+
+    return (
+        <Box display="flex" position="relative" width={1} height={1}>
+            <Box>
+                <Typography variant={"h5"} sx={{mb: 2}}>Integrasjonskonfigurasjon</Typography>
+                <form className={classes.form} onSubmit={onSubmit}>
+                    <Accordion expanded={expanded === 'panel1'} onChange={handleChange('panel1')} className={classes.accordion} defaultExpanded={true}>
```
Comment (fsjovatsen) [inline src/features/integration/IntegrationConfigurationForm.tsx:87]: Can this be simplified? Create a component? There is a lot of simular setup of each of them.

File: src/features/integration/components/form/ApplicantForm.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
+import {FormControl, FormGroup, InputLabel, MenuItem, Select, SelectChangeEvent, TextField} from '@mui/material';
+import React from 'react';
+import {dropdownPlaceholder} from "../../util/DefaultValues";
+
+const ApplicantForm: React.FunctionComponent<any> = (props) => {
+    return (
+        <FormGroup className={props.style.formControl}>
+            <FormControl>
```
Comment (fsjovatsen) [inline src/features/integration/components/form/ApplicantForm.tsx:8]: Similar setup on many components. Create a component for this?

File: src/features/integration/components/form/CaseForm.tsx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    SelectChangeEvent,
+    TextField
+} from '@mui/material';
+import React from 'react';
+import {dropdownPlaceholder} from "../../util/DefaultValues";
+
+const CaseForm: React.FunctionComponent<any> = (props) => {
+
+    return (
+        <div>
```
Comment (fsjovatsen) [inline src/features/integration/components/form/CaseForm.tsx:16]: Has the div a function? You could use `<>` instead.

File: src/features/integration/service/IntegrationService.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import http from "../../../http-common";
+import IFormData from "../types/Form/FormData";
+
```
Comment (fsjovatsen) [inline src/features/integration/service/IntegrationService.tsx:3]: Is this a repository and not a service?

File: src/features/main/Routes.ts [hunk 8]
```diff
@@ -13,17 +15,17 @@ const routes: IRoute[] = [
         inNavigationMenu: true
     },
     {
-        path: '/new_integration',
+        path: '/integration_configuration',
```
Comment (fsjovatsen) [inline src/features/main/Routes.ts:18]: See previous comment.

File: src/http-common.ts [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import axios from "axios";
+
+export default axios.create({
+    // TODO: change baseURL
+    baseURL: "http://localhost:8080/api",
```
Comment (fsjovatsen) [inline src/http-common.ts:5]: You should use the `proxy` config in `package.json` instead!

**Review by fsjovatsen — CHANGES_REQUESTED** [review]
Jævlig bra 👍 

Bare noen små kommentarer :)

File: src/features/integration/components/form/CaseForm.tsx (hunk 6 above)
Comment (ingrie) [reply src/features/integration/components/form/CaseForm.tsx:16]: removing unnecessary parent divs

File: src/features/integration/util/MapToDto.ts (hunk 3 above)
Comment (ingrie) [reply src/features/integration/util/MapToDto.ts:5]: changing to "toIntegrationConfiguration"

File: src/features/dashboard/Dashboard.tsx (hunk 1 above)
Comment (ingrie) [reply src/features/dashboard/Dashboard.tsx:28]: agree

File: src/http-common.ts (hunk 9 above)
Comment (ingrie) [reply src/http-common.ts:5]: will do 👍

File: src/features/integration/components/form/ApplicantForm.tsx [hunk 10]
```diff
@@ -1,56 +1,34 @@
-import {FormControl, FormGroup, InputLabel, MenuItem, Select, SelectChangeEvent, TextField} from '@mui/material';
+import {FormGroup} from '@mui/material';
 import React from 'react';
 import {dropdownPlaceholder} from "../../util/DefaultValues";
+import InputField from "./InputField";
+import {INPUT_TYPES} from "../../types/InputTypes.enum";
+import {IInputField} from "../../types/InputField";
 
 const ApplicantForm: React.FunctionComponent<any> = (props) => {
+    const inputTestArray: IInputField[] = [
```
Comment (ingrie) [inline src/features/integration/components/form/ApplicantForm.tsx:9]: will ofc change name

File: src/features/integration/components/form/ApplicantForm.tsx (hunk 5 above)
Comment (ingrie) [reply src/features/integration/components/form/ApplicantForm.tsx]: Inputfield component used in all form components

File: src/features/integration/IntegrationConfigurationForm.tsx (hunk 4 above)
Comment (ingrie) [reply src/features/integration/IntegrationConfigurationForm.tsx]: added a new accordioncompoenent

**Review by fsjovatsen — APPROVED** [review]

**Pull request merged**